src/main/java/hu/oe/nik/szfmv/automatedcar/model/utility/ModelCommonUtil.java

```diff
@@ -0,0 +1,42 @@
+package hu.oe.nik.szfmv.automatedcar.model.utility;
+
+import javax.imageio.ImageIO;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * Common utility functions in model
+ */
+public final class ModelCommonUtil {
+
+    private static final double MAX_DEGREE = 360;
+
+    /**
+     * Loads a png image from resources
+     * @param name Name of the image file
+     * @return A BufferedImage of the requested image
+     * @throws IOException If the IO operation cannot be processed, perhaps accessing rights
+     */
+    public static BufferedImage loadObjectImage(String name) throws IOException {
+        return ImageIO.read(new File(ClassLoader.getSystemResource((name + ".png")).getFile()));
+    }
+
+    /**
+     * Gets a rotation value from 4 points
+     * @param m11 Top left point
+     * @param m12 Top right point
+     * @param m21 Bottom left point
+     * @param m22 Bottom right point
+     * @return Double between [0,360[
+     */
+    public static double getRotationValue(double m11, double m12, double m21, double m22) {
+        // A Math.PI / 2.0 az elvileg a 0 pontot északra teszi, mert alapból kelet, én se értem
+        double angle = Math.atan2(m11 - m21, m12 - m22) + (Math.PI / 2.0E00);
+        angle = Math.toDegrees(angle);
+        if (angle < 0) {
+            return (angle + MAX_DEGREE);
+        }
+        return angle;
```

Comment:
Tamás példáját megnézted, hogy az, vagy ez jobb-e?

Reply:
Én sem számoltam végig, az m11, m12 stb. paraméterek elvileg egy forgatás mátrix értékei lennének:

| m11 | m12 |
| ------------- | ------------- |
| m21 | m22 |

és ebből kellene kiszámolnunk a forgatás mértékét.

Reply:
```Java
double angle = Math.atan2(m11, m21) + (Math.PI / 2.0E00);
```

A fogjuk az (1,0) vektor és elforgatjuk a mátrixunkkal akkor az elforgatott vektorból atan2()-val kiszámítható az x tengelytől a szög. Az (1,0) mátrix kelet felé van, így kell az észak felé forgatás is (már ha ragaszkodunk hozzá).